src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionService.kt

```diff
@@ -42,21 +54,83 @@ class IdentificationSessionService(private val identificationSessionRepository:
         return "${applicationProperties.baseUrl}$IDENTIFICATION_SESSIONS_BASE_PATH/${session.useIdSessionId}/$TCTOKEN_PATH_SUFFIX"
     }
 
+    @Transactional
```

Comment:
Does the whole function need to be transactional? We use that for the database, correct? I am wondering whether we could add this to a function that wraps the parts that are executing the database calls?